File: backend/internal/api/provider_keys.go
package api

import "strings"

type requestProvider struct {
	Provider     string
	OpenAIKey    string
	AnthropicKey string
}

func getRequestProviderHeaders(h func(string) string) requestProvider {
	p := strings.ToLower(strings.TrimSpace(h("X-AI-Provider")))
	if p != "anthropic" {
		p = "openai"
	}
	return requestProvider{
		Provider:     p,
		OpenAIKey:    strings.TrimSpace(h("X-OpenAI-Key")),
		AnthropicKey: strings.TrimSpace(h("X-Anthropic-Key")),
	}
}

File: frontend/src/services/providerKeys.ts
export type AIProvider = 'openai' | 'anthropic';

const LS_PROVIDER = 'ai_provider';
const LS_OPENAI_KEY = 'openai_api_key';
const LS_ANTHROPIC_KEY = 'anthropic_api_key';

export type ProviderKeys = {
  provider: AIProvider;
  openAIKey?: string;
  anthropicKey?: string;
};

function safeRead(key: string): string {
  try {
    return localStorage.getItem(key) || '';
  } catch {
    return '';
  }
}

export function getProviderKeys(): ProviderKeys {
  const providerRaw = safeRead(LS_PROVIDER);
  const provider: AIProvider = providerRaw === 'anthropic' ? 'anthropic' : 'openai';
  const openAIKey = safeRead(LS_OPENAI_KEY).trim();
  const anthropicKey = safeRead(LS_ANTHROPIC_KEY).trim();

  return {
    provider,
    openAIKey: openAIKey || undefined,
    anthropicKey: anthropicKey || undefined,
  };
}
